### ISSUE TITLE
Feature: Exactly-once semantics

### ISSUE THREAD

**Issue by Psykepro** [lovoo/goka#321]
Is the exactly-once semantics added to the road map of the library? Would be very nice to have it. As it is the most wanted semantics in the stream processing realm.

**Comment by frairon** [comment]
Absolutely, I understand the necessity of this. It's simply a matter of time and ressources, so we have to evaluate how easy/fast it is to add this to the framework and if it's worth the effort. Let me check in the next days (or feel free to give it a look too), but I can't promise anything.

**Comment by twmb** [comment]
As a little bit of self promotion, my [franz-go](https://github.com/twmb/franz-go) client has EOS support, if the problem for supporting EOS in goka is due to sarama's lack of EOS support.

**Comment by tinkalgogoi** [comment]
Currently, Sarama added transactional API from version [v1.37.2](https://github.com/Shopify/sarama/releases/tag/v1.37.2), Here is a good article showing the implementation of transactions in producer/consumer - https://www.andreamedda.com/posts/play-with-sarama-transactional-api
